### PR TITLE
Add registration conformation email

### LINKED ISSUE #2310: Get confirmation mail when registering for events
### Describe the solution you'd like
An option in your profile to get confirmation emails when registering for event, these emails should have the date and such in them so that Google/Apple can recognise them and add them to your calendar automatically.

### Motivation
This way people can easily see and remember the events they have registered for.

### PR DESCRIPTION
**Pull request by T8902**
Closes #2310

### Summary
Added option to get registrations conformations via email in the profile  and if it is turned on 
users will receive a conformation of registration via email.

### How to test
Steps to test the changes you made:
1.  Make sure the check mark to receive conformation emails is on in profile
2.  Go to Calendar on the site
3.  Click on Register
4.  Turn the check-mark off
5. register again for an event

### CODE REVIEW

File: website/events/templates/events/email/registration_queued_email.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{% extends "email/html_email.html" %}
+{% block content %}
+<p>Dear {{ registration.member.profile.display_name|safe }},</p>
```
Comment (DeD1rk) [inline website/events/templates/events/email/registration_queued_email.html:3]:
```suggestion
<p>Dear {{ registration.member.profile.display_name }},</p>
```

File: website/events/templates/events/email/registration_conf_email.html [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{% extends "email/html_email.html" %}
+{% block content %}
+<p>Dear {{ registration.member.profile.display_name }},</p>
```
Comment (DeD1rk) [inline website/events/templates/events/email/registration_conf_email.html:3]:
```suggestion
<p>Dear {{ name }},</p>
```

Since member may be None, probably easiest is to determine a name outside of the email template and pass it as context.

File: website/events/templates/events/email/registration_queued_email.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{% extends "email/html_email.html" %}
+{% block content %}
+  <p>Dear {{ registration.member.profile.display_name }},</p>
+
+  <p>Hereby a confirmation that you registered for the following even: '{{ event.title }}', on:  {{ registration.date }}.</p>
+  <p>Since this event is full right now you have been put on the queue on position: {{ registration.queue_position }}.</p>
+  <p>You will be contacted when you get a spot to participate in the event.</p>
+  {% if event.cancel_deadline %}
+  <p>If you would like to cancel this registration please do so before {{ event.cancel_deadline }}.</p>
```
Comment (DeD1rk) [inline website/events/templates/events/email/registration_queued_email.html:9]: Format deadline

File: website/events/templates/events/email/registration_conf_email.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+
+<p>Hereby a confirmation that you registered for the following even: '{{ event.title }}', on:  {{ registration.date }}.</p>
+<p>Please inform the website on any other possible information needed from you to complete your registration.</p>
+{% if event.cancel_deadline %}
+<p>If you would like to cancel this registration please do so before {{ event.cancel_deadline }}.</p>
```
Comment (DeD1rk) [inline website/events/templates/events/email/registration_conf_email.html:8]: Format deadline

Comment (DeD1rk) [reply website/events/templates/events/email/registration_conf_email.html:8]: And indentation here

**Review by DeD1rk — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -18,7 +18,16 @@ The latest Thalia Website built on Django.
...
+## Useful git commands
+
+- `git push` to push your branch to github
+- `git checkout -b <branch name>` to create a branch to work on
+- `git fetch origin` to synchronize your local repository with the central repository's main branch
+- `git switch <branch name>` to switch between branches
+- `git add .` to add all your changes to a commit (mind to run `make fmt` in your terminal before adding changes)
+- `git commit (-m <commit message>)` to commit your added changes possibly with a message
+- `git rebase origin/master` to get your branch up to date with all merges
+- `git status` to check the current status of your branch
```
Comment (JobDoesburg) [inline README.md:30]: This should really be in a separate PR, sorry

File: website/events/templates/events/email/registration_conf_email.html [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{% extends "email/html_email.html" %}
+{% block content %}
+  <p>Dear {{ registration.member.profile.display_name }},</p>
+
+  <p>Hereby a confirmation that you registered for the following even: '{{ event.title }}', on:  {{ registration.date }}.</p>
```
Comment (DeD1rk) [inline website/events/templates/events/email/registration_conf_email.html:5]: format

File: website/events/templates/events/email/registration_conf_email.html [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+  <p>Please inform the website on any other possible information needed from you to complete your registration.</p>
```
Comment (DeD1rk) [inline website/events/templates/events/email/registration_conf_email.html:6]: This line isn't needed when the event has no registration fields (that should be easy to check)

File: website/events/templates/events/email/registration_queued_email.txt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Dear {{ registration.member.profile.display_name|safe }},
```
Comment (DeD1rk) [inline website/events/templates/events/email/registration_queued_email.txt:1]: It might be good to add a closing to the txt emails too (the html parent template adds it automatically, but txt has no parent template.

**Comment by DeD1rk** [comment]
I refactored some stuff and tried it out. I think it should work properly like this. But haven't written tests of course 😅

**Review by DeD1rk — APPROVED** [review]
